app/services/orders/handle_fees_service.rb

```diff
@@ -58,6 +64,57 @@ def update_order_fees!
 
     private
 
+    def create_line_item_fees!(line_item)
+      return unless provided_by_order_cycle? line_item
+
+      calculator.create_line_item_adjustments_for(line_item)
+    end
+
+    def create_or_update_line_item_fee!(line_item)
+      applicators = calculator.per_item_enterprise_fee_applicators_for(line_item.variant)
+
+      applicators.each do |fee_applicator|
+        fee_adjustment = line_item.adjustments.find_by(originator: fee_applicator.enterprise_fee)
+
+        if fee_adjustment
+          fee_adjustment.update_adjustment!(line_item, force: true)
+        elsif provided_by_order_cycle? line_item
+          fee_applicator.create_line_item_adjustment(line_item)
+        end
+      end
+
+      # Update any fees not already processed
+      fees_to_update =
+        order_cycle_per_item_enterprise_fee_for - applicators.map(&:enterprise_fee)
+      fees_to_update.each do |fee|
+        fee_adjustment = line_item.adjustments.find_by(originator: fee)
+
+        fee_adjustment&.update_adjustment!(line_item, force: true)
+      end
+    end
+
+    def delete_removed_fees!(line_item)
+      order_cycle_fees = order_cycle_per_item_enterprise_fee_for
+
+      removed_fees = line_item.enterprise_fee_adjustments.where.not(originator: order_cycle_fees)
+
+      removed_fees.each(&:destroy)
+    end
+
+    def order_cycle_per_item_enterprise_fee_for
```

Comment:
"for..."?  This method returns an array of per_item enterprise_fees, so how about:
```suggestion
    def order_cycle_per_item_enterprise_fees
```

(Ah, I see this method was copied from `per_item_enterprise_fee_applicators_for(variant)`)